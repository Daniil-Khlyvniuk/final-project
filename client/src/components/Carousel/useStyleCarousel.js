import { makeStyles } from '@mui/styles'
import theme from '../../utils/Theme'

export const useStyleCarousel = makeStyles({
	slide:{
		width: '1180px',
		height: '630px',
		position: 'relative',

	},
	arrowLeft:{
		width: '25px',
		height: '25px',
		borderTop: '2px solid #5C5E60',
		borderRight: '2px solid #5C5E60',
		marginRight: '60px',
		transform: 'rotate(-135deg)'
	},
	arrowRight: {
		width: '25px',
		height: '25px',
		borderTop: '2px solid #5C5E60',
		borderRight: '2px solid #5C5E60',
		marginRight: '60px',
		transform: 'rotate(45deg)'
	},
	nextEl: {
		position: 'absolute',
		top:'50%',
		right:'-3%',
		transform:'translate(-50%, -50%)',
		background: '#FFFFFF',
		borderRadius: '2px',
		padding: '25px 24px 25px 0',
		width: '20px',
		zIndex: 10,
		cursor: 'pointer'
	},
	prevEl: {
		position: 'absolute',
		top:'50%',
		left:'1%',
		transform:'translate(-50%, -50%)',
		background: '#FFFFFF',
		borderRadius: '2px',
		padding: '25px 4px 25px 22px',
		width: '20px',
		zIndex: 10,
		cursor: 'pointer'
	},
	textBlock: {
		position:'absolute',
		bottom: '100px',
		left: '50px'
	},
	pagination:{
		marginTop: '20px',
		'& .swiper':{
			borderRadius: '4px',
		},
		'& .swiper-pagination-horizontal':{
			bottom:'1px',
			background: '#fff',
			// position: 'fixed',
			'& .swiper-pagination-bullet':{
				margin: 0
			}
		},
		'& .swiper-pagination-bullet':{
			width: '15%',
			height:  '2px',
			background: 'black',
			borderRadius: '0'
		},
		'& .swiper-pagination-bullet-active':{
			'&::before':{
				content: '""',
				display: 'block',
				height: '2px',
				width: '100%',
				borderBottom: '3px solid #000',
			}
		}
	},
	title: {
		display: 'inline-block',
		fontFamily: theme.typography.h2.fontFamily,
		fontStyle: 'normal',
		fontWeight: 'normal',
		fontSize: 32,
		lineHeight: '50px',
		color: '#FFFFFF',
		backgroundColor: '#373F41',
		paddingTop: '5px',
		paddingBottom: '5px',
		paddingLeft: '8px',
		paddingRight: '8px',
		borderRadius: '2px',
		marginBottom: '-13px'
	},
	desc:{
		fontFamily: theme.typography.h2.fontFamily,
		fontStyle: 'normal',
		fontWeight: 'normal',
		fontSize: '16px',
		lineHeight: '18px',
		letterSpacing: '0.02em',
		color: '#FFFFFF',
		textShadow: '0px 1px 1px rgba(0, 0, 0, 0.25)',
		backgroundColor:'#373F41',
		padding: '20px'
	},
	ads:{

	},
	shopBtn:{
		cursor: 'pointer',
		width: '192px',
		height: '58px',
		background: '#FFFFFF',
		border: 'none',
		boxShadow: '0px 1px 1px rgba(0, 0, 0, 0.45)',
		borderRadius: '4px',
		fontFamily: theme.typography.h2.fontFamily,
		fontStyle: 'normal',
		fontWeight: 'normal',
		fontSize: 16,
		lineHeight: '32px',
		color: '#373F41'
	}
})