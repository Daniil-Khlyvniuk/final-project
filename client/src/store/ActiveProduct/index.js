
export { default as activeProductSelector } from './selectors'
export { default as activeProductOperations } from './operations'
import activeProductReducer from './activeProductSlice'

export default activeProductReducer