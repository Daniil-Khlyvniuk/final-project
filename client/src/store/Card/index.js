export { default as cardSelectors } from './selectors'
export { default as cardOperations } from './operations'
import * as cardActions from './cardSlice'

export default cardActions


