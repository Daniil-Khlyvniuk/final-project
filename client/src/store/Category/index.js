export { default as categorySelectors } from './selectors'
export { default as categoryOperations } from './operations'

import * as categoryActions from './categorySlice'
export default categoryActions