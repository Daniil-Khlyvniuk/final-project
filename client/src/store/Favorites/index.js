export { default as favoritesSelectors } from './selectors'
export { default as favoritesOperations } from './operations'

import * as favoritesActions from './favoritesSlice'
export default favoritesActions