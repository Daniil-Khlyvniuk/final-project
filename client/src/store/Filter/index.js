import reducer from './filterSlice'

export {default as filterSelectors} from './selectors'
export {default as filterOperations} from './operations'

export default reducer