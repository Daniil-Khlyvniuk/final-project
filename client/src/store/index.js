import { configureStore } from '@reduxjs/toolkit'
import productsSlice from './Products/productsSlice'
import linksSlice from './Links/'
import categorySlice from './Category/categorySlice'
import sliderSlice from './Slider'
import modalSlice from './Modal/modalSlice'
import shoppingBagSlice from './ShoppingBag/shoppingBagSlice'
import productSlice from './Product/productSlice'
import filterSlice from './Filter'

import userSlice from './User'

const store = configureStore({
	reducer: {
		slides: sliderSlice,
		products: productsSlice,
		links: linksSlice,
		category: categorySlice,
		modal: modalSlice,
		shoppingBag: shoppingBagSlice,
		user: userSlice,
		product : productSlice,
		filter: filterSlice,

	},
	middleware: (getDefaultMiddleware) =>
		getDefaultMiddleware({
			serializableCheck: {
				// Ignore these action types
				ignoredActions: ['modal/modalToggle', 'user/setToken', 'user/fetchUser/pending','user/fetchUser/fulfilled','filter/setFiltersFromQueryString'],
			},
		}),
})

export default store
