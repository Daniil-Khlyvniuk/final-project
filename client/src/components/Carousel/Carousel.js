import React, { useEffect, useRef, useState } from 'react'
import PropTypes from 'prop-types'
import Slider from 'react-slick'
// import 'slick-carousel/slick/slick.css'
// import 'slick-carousel/slick/slick-theme.css'
import './slick(costil).css'
import './slick-theme(costil).css'
import { useStyleCarousel } from '../../utils/customHooks/useStyleCarousel'

const Carousel = ({
	slides,
	main,
	product,
	related  }) => {
	const style = useStyleCarousel()
	const [nav1, setNav1] = useState()
	const [nav2, setNav2] = useState()
	const slider = useRef()
	const thumbs = useRef()

	// eslint-disable-next-line no-console
	console.log(slides)

	useEffect(() => {
		setNav1(slider.current)
		setNav2(thumbs.current)
	}, [])


	const SampleNextArrow = (props) => {
		const { className, onClick } = props
		return (
			<div className={style.nextEl}>
				<div
					className={className + style.nextEl}
					style={{
						borderTop: '2px solid #5C5E60',
						borderRight: '2px solid #5C5E60',
						transform: 'rotate(45deg)'
					}}
					onClick={onClick}
				/>
			</div>
		)
	}

	const  SamplePrevArrow = (props) => {
		const { className, onClick } = props
		return (
			<div className={style.prevEl}>
				<div
					className={className + style.prevEl}
					style={{
						borderTop: '2px solid #5C5E60',
						borderRight: '2px solid #5C5E60',
						transform: 'rotate(-135deg)' }}
					onClick={onClick}
				/>
			</div>
		)
	}

	const settingsMain = {
		dots: true,
		infinite: true,
		fade: true,
		speed: 500,
		arrows: true,
		slidesToShow: 1,
		slidesToScroll: 1,
		nextArrow: <SampleNextArrow/>,
		prevArrow: <SamplePrevArrow/>,
	}
	const settingRelated = {
		infinite: true,
		speed: 500,
		slidesToShow: 3,
		initialSlide: 1,
		arrows: true,
		nextArrow: <SampleNextArrow/>,
		prevArrow: <SamplePrevArrow/>,
	}
	const settingsProducts = {
		slidesToShow: 1,
		slidesToScroll: 1,
		fade: true,
		arrows: true,
		nextArrow: <SampleNextArrow/>,
		prevArrow: <SamplePrevArrow/>,
	}
	const settingThumbs = {
		slidesToShow: 3,
		slidesToScroll: 1,
		initialSlide: 1,
		centerMode: true,
		focusOnSelect: true,
	}


	return (
		<>
			{main &&
			<Slider
				{...settingsMain}
				className={style.show}
			>
				{slides?.map((slide) => {
					return (
						<div key={slide.customId} className={style.slideContainer}>
							<img src={slide.imageUrl} className={style.slide} alt=""/>
							<div className={style.textBlock}>
								<p className={style.title}>{slide.title}</p>
								{slide.description &&
								<p className={style.desc}>{slide.description}</p>}
							</div>
						</div>
					)
				})}
			</Slider>
			}

			{product &&
				<>
					<div style={{width: '50%'}}>
						<Slider
							{...settingsProducts}
							asNavFor={nav2}
							ref={slider}
						>
							{slides?.map((slide, index) => {
								// eslint-disable-next-line no-console
								console.log('slide_1', slide)
								return (
									<div key={index} className={style.slideContainer}>
										<img src={`http://localhost:5000/${slide}`} className={style.slide} alt=""/>
									</div>
								)
							})}
						</Slider>
						<Slider
							{...settingThumbs}
							asNavFor={nav1}
							ref={thumbs}
							className={style.thumbWrapper}
						>
							{slides?.map((slide, index) => {
								// eslint-disable-next-line no-console
								console.log('slide_2', slide)
								return (
									<div key={index}>
										<img src={`http://localhost:5000/${slide}`} className={style.thumb} alt=""/>
									</div>
								)
							})}
						</Slider>
					</div>
				</>
			}

			{related &&
				<>
					<h3 className={style.relatedTitle}>RELATED ITEMS</h3>
					<Slider
						{...settingRelated}
					>
						{slides?.map((slide) => {
							return (
								<div key={slide.customId}>
									<img src={slide.imageUrl} className={style.thumb} alt=""/>
								</div>
							)
						})}
					</Slider>
				</>
			}
		</>
	)
}

Carousel.propTypes = {
	isLoading: PropTypes.bool,
	slides: PropTypes.array,
	main: PropTypes.bool,
	related: PropTypes.bool,
	product: PropTypes.bool,
	className: PropTypes.string,
	onClick: PropTypes.func
}

export default Carousel