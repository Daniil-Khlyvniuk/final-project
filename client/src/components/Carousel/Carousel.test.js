import React from 'react'
// eslint-disable-next-line no-unused-vars
import { matchMedia } from '../../tests/matchMedia'
import Carousel from './Carousel'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import store from '../../store'
import { BrowserRouter } from 'react-router-dom'
import { ErrorMessage } from 'formik'

const slides = []
const main = true
const product = true
const related = true
// const setUp = (props) => render(<Carousel {...props}/>)

describe('Test Carousel component', () => {
	test('Smoke test Carousel', () => {
		render(
			<Provider store={store}>
				<BrowserRouter>
					<Carousel
						slides={slides}
						main={main}
						product={product}
						related={related}
					/>
				</BrowserRouter>
			</Provider>
		)
	})

	test('Carousel Snapshot test', () => {
		const container = render(
			<Provider store={store}>
				<BrowserRouter>
					<Carousel
						slides={slides}
						main={main}
						product={product}
						related={related}
					/>
				</BrowserRouter>
			</Provider>
		)
		expect(container).toMatchSnapshot()
	})

})

