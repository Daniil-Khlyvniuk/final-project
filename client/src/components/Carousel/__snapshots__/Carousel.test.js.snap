// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Carousel component Carousel Snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="slick-slider makeStyles-show-37 slick-initialized"
      >
        
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          />
        </div>
        
        
      </div>
      <div
        class="makeStyles-productsWrapper-40"
      >
        <div
          class="slick-slider makeStyles-products-39 slick-initialized"
        >
          
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1;"
            />
          </div>
          
          
        </div>
        <div
          class="slick-slider makeStyles-thumbWrapper-35 slick-initialized"
        >
          
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px); transition: transform 500ms ease;"
            />
          </div>
          
          
        </div>
      </div>
      <h3
        class="makeStyles-relatedTitle-41"
      >
        RELATED ITEMS
      </h3>
      <div
        class="slick-slider makeStyles-related-38 slick-initialized"
      >
        
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          />
        </div>
        
        
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="slick-slider makeStyles-show-37 slick-initialized"
    >
      
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        />
      </div>
      
      
    </div>
    <div
      class="makeStyles-productsWrapper-40"
    >
      <div
        class="slick-slider makeStyles-products-39 slick-initialized"
      >
        
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1;"
          />
        </div>
        
        
      </div>
      <div
        class="slick-slider makeStyles-thumbWrapper-35 slick-initialized"
      >
        
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px); transition: transform 500ms ease;"
          />
        </div>
        
        
      </div>
    </div>
    <h3
      class="makeStyles-relatedTitle-41"
    >
      RELATED ITEMS
    </h3>
    <div
      class="slick-slider makeStyles-related-38 slick-initialized"
    >
      
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        />
      </div>
      
      
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
